_posts/2020-05-02-schedule-your-functions.md

```diff
@@ -0,0 +1,264 @@
+    created_date = datetime(created_date[0], created_date[1], created_date[2])
+    return datetime.now() - created_date
+
+
+def lock_issue(issue_number, auth_header, username, repo):
+    r = requests.put(
+        url=f"https://api.github.com/repos/{username}/{repo}/issues/{issue_number}/lock", headers=auth_header)
+    if r.status_code == 204:
+        return f"Issue: {issue_number} locked"
+
+
+def get_authorization_header():
+    with open("/var/openfaas/secrets/auth-token") as file:
+        return {'Authorization': f"token {file.read().strip()}"}
+
+
+def handle(req):
+    username = os.getenv("github_username")
+    repo = os.getenv("github_repository")
+    days_before_outdated = int(os.getenv("days_before_outdated"))
+    r = requests.get(f"https://api.github.com/repos/{username}/{repo}/issues")
+    issues = r.json()
+
+    issue_numbers = []
+    for issue in issues:
+        if not issue.get('pull_request'):
+            diff = get_difference_in_days(issue.get('created_at'))
+            if diff.days > days_before_outdated and not issue.get("locked"):
+                issue_numbers.append(int(issue.get('number')))
+
+    locked_issues = ""
+    if len(issue_numbers) > 0:
+        auth_header = get_authorization_header()
+        for issue_number in issue_numbers:
+            locked_issues = locked_issues + \
+                f"{lock_issue(issue_number, auth_header, username, repo)}\n"
+
+    if locked_issues != "":
+        return locked_issues
+
+    return "No issues locked"
+```
+
+In general the function will fetch the issues from the repository you chose in the configuration section then it will filter them and calculate, using the date when they were opened, whether they are older than the 90 days we chose. If this is the case the lockbot will authenticate itself against the GitHub API with the access token we generated and will lock the issue from further conversation. The lockbot will respond with the locked issues or the lack of such.
+
+### Deploy the function
+
+In order to actually make our lockbot start watching the repository we will deploy it with the following command:
+
+```bash
+faas up
+```
+
+A full working example of the function can be see in the [lockbot](https://github.com/martindekov/lockbot) repository.
+
+>Note: If you are eager to see the function in action you can directly invoke it using the UI or the CLI.
+
+## Wrapping up
+
+In this post we have shown how you can run minimal OpenFaaS environment with simple function and how with the help of the cron-connector you can execute the function on a desired schedule.
+
+Learn how to create your own functions with the [OpenFaaS workshop](https://github.com/openfaas/workshop). We used one of the templates from the store in our example, but you'll find many others, just run `faas-cli template store list` or view the [Template Documentation](https://docs.openfaas.com/cli/templates/) and make sure your functions always run on time!
```

Comment:
Let's have a link for the triggers page in the documentation?

Reply:
Will add :+1:

Reply:
Added link after the workshop :heavy_check_mark: